frontend/app/custom-components/profile/edit/topics.tsx

```diff
@@ -26,9 +29,74 @@ export function Topics({ profile, setProfile }: TopicsProps) {
     profile.topics.length > 0 ? profile.topics[0].name : undefined,
   );
   const [newTopicName, setNewTopicName] = useState("");
-  const selectedTopicKeywords =
-    profile.topics.find((topic) => topic.name === selectedTopic)?.keywords ||
-    [];
+  const [aiSuggestions, setAiSuggestions] = useState<string[]>([]);
+  const [isLoadingSuggestions, setIsLoadingSuggestions] = useState(false);
+
+  const selectedTopicKeywords = useMemo(() => {
+    return (
+      profile.topics.find((topic) => topic.name === selectedTopic)?.keywords ||
+      []
+    );
+  }, [profile.topics, selectedTopic]);
+
+  const { authorizationHeaders } = useAuthorization();
+
+  const getSuggestions = async (keywords: string[]) => {
+    if (keywords.length === 0) {
+      setAiSuggestions([]);
+      return;
+    }
+
+    setIsLoadingSuggestions(true);
+    try {
+      const result = await client.POST(
+        "/api/v1/search-profiles/keywords/suggestions",
+        {
+          body: keywords,
+          headers: authorizationHeaders,
+        },
+      );
+
+      if (result.data?.suggestions) {
+        const filteredSuggestions = result.data.suggestions.filter(
+          (suggestion: string) =>
+            !keywords.some(
+              (keyword) => keyword.toLowerCase() === suggestion.toLowerCase(),
+            ),
+        );
+        setAiSuggestions(filteredSuggestions);
+      }
+    } catch (error) {
+      console.error("Failed to fetch AI suggestions:", error);
+      toast.error("Failed to get AI suggestions.");
+      setAiSuggestions([]);
+    } finally {
+      setIsLoadingSuggestions(false);
+    }
+  };
+
+  const keywordsString = selectedTopicKeywords.join(",");
+
+  useEffect(() => {
+    if (selectedTopic && selectedTopicKeywords.length > 0) {
+      getSuggestions(selectedTopicKeywords);
+    } else {
+      setAiSuggestions([]);
+    }
+  }, [selectedTopic, keywordsString]);
```

Comment:
Why cant we do it like this?

```suggestion

  useEffect(() => {
    if (selectedTopic && selectedTopicKeywords.length > 0) {
      getSuggestions(selectedTopicKeywords);
    } else {
      setAiSuggestions([]);
    }
  }, [selectedTopic, selectedTopicKeywords]);
```